app/pages/sections/DocumentsSection.scala

```diff
@@ -18,15 +18,18 @@ package pages.sections
 
 import controllers.routes
 import models.{Mode, UserAnswers}
-import play.api.libs.json.{JsArray, JsPath}
+import pages.{AddAnotherDocumentPage, AddAnotherPage}
+import play.api.libs.json.JsPath
 import play.api.mvc.Call
 
-case object DocumentsSection extends Section[JsArray] {
+case object DocumentsSection extends AddAnotherSection {
 
   override def path: JsPath = ParentSection.path \ toString
 
   override def toString: String = "documents"
 
   override def route(userAnswers: UserAnswers, mode: Mode): Option[Call] =
     Some(routes.AddAnotherDocumentController.onPageLoad(userAnswers.lrn))
+
```

Comment:
Not major but the `override def route` lines can be removed as this now no longer changes the value derived in the AddAnotherSection trait 🙂